pkg/bgpv1/agent/controller.go

```diff
@@ -159,13 +160,24 @@ func configureForClusterPoolIPAM(factory externalversions.SharedInformerFactory,
 		UpdateFunc: func(_ interface{}, _ interface{}) { sig.Event(struct{}{}) },
 		DeleteFunc: sig.Event,
 	})
-	go nodeInformer.Run(stop)
 	return &ciliumNodeSpecer{
-		Name:       name,
-		NodeLister: nodeLister,
+		Name:         name,
+		NodeLister:   nodeLister,
+		nodeInformer: nodeInformer,
 	}, nil
 }
 
+// ControllerParams contains all parameters needed to construct a Controller
+type ControllerParams struct {
+	cell.In
+
+	Lifecycle    hive.Lifecycle
+	Clientset    client.Clientset
+	RouteMgr     BGPRouterManager
+	DaemonConfig *option.DaemonConfig
```

Comment:
Hmm, just thinking out loud:

Can we already introduce a BGPConfig (without Flags()) and do a projection from DaemonConfig into it?
Or since we have the `ControllerOpt` stuff already, could we create a set of those from `DaemonConfig`?
E.g. try to avoid exposing the whole DaemonConfig into the innards of modules.

Reply:
As discussed offline, we can completely remove the `EnableBGPControlPlane` from the DaemonConfig and handle it via the cell.Config. Would like to do that in the followup PR since this PR still uses the `IPAM` field as well, which is no longer required in the next PR.